### PR TITLE
Suggest migrating to DistillationTrainer when the GKD config is fully covered

### PR DESCRIPTION
**Pull request by qgallouedec**
experimental.gkd.GKDTrainer now logs a warning when its config is exactly reproducible with the main-codebase DistillationTrainer, i.e. lmbda=1.0 and temperature=1.0:

lmbda=1.0 → fully on-policy, the only mode DistillationTrainer supports (it also makes the seq_kd branch dead code).
temperature=1.0 → GKD applies temperature to generation only, DistillationTrainer applies it to generation and the divergence; at any other value the losses differ.
Silent for the GKD defaults (lmbda=0.5, temperature=0.9), deliberately narrow.

---

> [!NOTE]
> **Low Risk**
> Adds an init-time warning only; no training logic, defaults, or loss behavior changes.
> 
> **Overview**
> **`GKDTrainer`** now emits a **runtime warning** when **`lmbda=1.0`** and **`temperature=1.0`**, because that combination matches what the maintained **`DistillationTrainer`** already does (fully on-policy; divergence temperature aligned with generation). The message points users to **`DistillationConfig`/`DistillationTrainer`**, renaming **`max_new_tokens`** → **`max_completion_length`**, and setting **`beta`** explicitly (defaults differ: 0.5 vs 1.0). It also notes **`DistillationTrainer`** extras: vLLM generation and memory-efficient chunked loss.
> 
> Default GKD settings (**`lmbda=0.5`**, **`temperature=0.9`**) are unchanged and stay silent. Logging uses **`accelerate.logging.get_logger`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3c4fba9a287b614bbc65be4a304b8c56dcae2bde. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/gkd/gkd_trainer.py [hunk 1]
```diff
@@ -213,6 +217,18 @@ def __init__(
...
+        # With `lmbda=1.0` training is fully on-policy and `seq_kd` is never reached, and with `temperature=1.0` the
+        # sampling temperature matches the one `DistillationTrainer` also applies to the divergence. In that setting,
+        # `DistillationTrainer` covers this run and additionally supports vLLM generation and a chunked loss that
+        # never materializes the full logits.
+        if self.lmbda == 1.0 and self.temperature == 1.0:
+            logger.warning(
+                "This GKD configuration (`lmbda=1.0`, `temperature=1.0`) is fully covered by `DistillationTrainer`, "
+                "which is maintained in the main codebase and supports vLLM generation and a memory-efficient "
+                "chunked loss. Consider migrating: replace `GKDConfig`/`GKDTrainer` with "
+                "`DistillationConfig`/`DistillationTrainer` and `max_new_tokens` with `max_completion_length`."
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/gkd/gkd_trainer.py]: Correct, fixed in 09c20154: the warning now tells the user to set beta explicitly and prints their current value (same semantics in both configs, defaults 0.5 vs 1.0).

**Comment by qgallouedec** [comment]
Merging without review, it's only a warning, so nothing breaking for the user, feel free to give late review if you think I'm missing something.
